extras/tools/delete_unowned_bricks.sh

```diff
@@ -0,0 +1,288 @@
+            shift
+            scan_bricks "$@"
+        ;;
+        delete)
+            shift
+            delete_bricks "$@"
+        ;;
+        summarize)
+            shift
+            summarize_bricks "$@"
+        ;;
+        *)
+            usage
+            echo "error: must specify either 'scan' or 'delete'" >&2
+            exit 1
+        ;;
+    esac
+}
+
+get_ip(){
+    if [[ -z "$ip_prefix" ]]; then
+        echo "error: ip_prefix is required" >&2
+        exit 2
+    fi
+
+    matches=$(ip a | grep 'inet\b' | grep -c "$ip_prefix")
+    if [[ "$matches" -gt 1 ]]
+    then
+        echo "error: too many ips matching prefix $ip_prefix found." >&2
+        exit 1
+    fi
+    if [[ "$matches" -eq 0 ]]
+    then
+        echo "error: no ips matching prefix $ip_prefix found." >&2
+        exit 1
+    fi
+    ip a | grep 'inet\b' | grep "$ip_prefix" | awk '{print $2}' | cut -d"/" -f1
+}
+
+scan_bricks() {
+    parse_args "$@"
+
+    if [[ -z "$fstab_path" ]]; then
+        echo "error: fstab path is required" >&2
+        exit 2
+    fi
+
+    ip=$(get_ip)
+    if [[ -z "$ip" ]]; then
+        exit 0
+    fi
+
+    allvgs=$(vgs | grep vg_ | awk '{print $1}'  | awk -F '_' '{print $2}')
+
+    for eachVG in $allvgs
+    do
+        gluster volume info | grep "Brick[0-9]" | grep "$ip" | grep "$eachVG" | awk '{print $2}'  | awk -F":" '{print $2}' | awk -F'/' '{print $7}' | awk -F'_' '{print $2}' | sort -u > "gluster_bricks_in_${eachVG}.txt"
+    done
+
+    for eachVG in $allvgs
+    do
+        lvs | grep brick_  | grep "$eachVG" |  awk '{print $1}' | awk  -F '_' '{print $2}' | sort -u > "All_in_${eachVG}.txt"
+    done
+
+    for eachVG in $allvgs
+    do
+        sdiff "gluster_bricks_in_${eachVG}.txt" "All_in_${eachVG}.txt" | grep \> | awk '{print $2}' > "to_delete_${eachVG}.txt"
+        echo "./delete_unowned_bricks.sh delete --ip-prefix=${ip_prefix} --fstab-path=${fstab_path} --vgid=${eachVG} --lvlist=to_delete_${eachVG}.txt --prompt"
+    done
+
+    rm -f gluster_bricks_in_*
+    rm -f All_in_*
+}
+
+summarize_bricks() {
+    parse_args "$@"
+
+    if [[ -z "$fstab_path" ]]; then
+        echo "error: fstab path is required" >&2
+        exit 2
+    fi
+
+    ip=$(get_ip)
+    if [[ -z "$ip" ]]; then
+        exit 0
+    fi
+
+    summary "$ip"
+}
+
+
+
+
+summary() {
+    echo "Mount count"
+    mount | grep -c brick_
+    echo "bricks in fstab count"
+    grep -c brick_ "$fstab_path"
+    echo "lvs count"
+    lvs | grep -c brick_
+    echo "gluster bricks count"
+    gluster volume info | grep "^Brick[0-9]" | grep -c "$1"
+}
+
+
+delete_bricks() {
+
+    parse_args "$@"
+
+    if [[ -z "$vg_id" ]]; then
+        echo "error: vgid is required" >&2
+        exit 2
+    fi
+
+    if [[ -z "$fstab_path" ]]; then
+        echo "error: fstab path is required" >&2
+        exit 2
+    fi
+
+    echo ""
+    echo "Checking for glusterd on local and peer nodes"
+    checkLocalNodeHasGlusterdRunning
+    #checkPeerNodeStatus
```

Comment:
This function is not used because it is not guaranteed that we would have ssh access to other gluster nodes. The script acts only locally and has to be run on all the nodes one after other. Will delete references to peer actions everywhere.